protocols/identify/src/handler.rs

```diff
@@ -371,8 +400,18 @@ impl ConnectionHandler for Handler {
             ConnectionEvent::FullyNegotiatedOutbound(fully_negotiated_outbound) => {
                 self.on_fully_negotiated_outbound(fully_negotiated_outbound)
             }
-            ConnectionEvent::DialUpgradeError(dial_upgrade_error) => {
-                self.on_dial_upgrade_error(dial_upgrade_error)
+            ConnectionEvent::DialUpgradeError(DialUpgradeError { error, .. }) => {
+                let upgrade_error = match error {
+                    StreamUpgradeError::Timeout => StreamUpgradeError::Timeout,
+                    StreamUpgradeError::NegotiationFailed => StreamUpgradeError::NegotiationFailed,
+                    StreamUpgradeError::Io(e) => StreamUpgradeError::Io(e),
+                    StreamUpgradeError::Apply(v) => unreachable!("{}", v),
```

Comment:
```suggestion
                    StreamUpgradeError::Apply(v) => void::unreachable!(v),
```

Shouldn't this static check work?

Reply:
We can't do this because of `v` is `Either<Void, Void>`

Reply:
You can call `into_inner` on `v` if they are both the same!